### PR TITLE
More setups add custom grub entries

### PR DESCRIPTION
**Pull request by pevik**
Verification run:
* http://quasar.suse.cz/tests/4532 (`GRUB_PARAM=ima_policy=tcb;debug_pagealloc=on;foo=bar foo2=bar2`)
and test which runs it: has 3 sections `ima_policy=tcb`, `debug_pagealloc=on` and `foo=bar foo2=bar2`
http://quasar.suse.cz/tests/4537#step/boot_ltp/2

* http://quasar.suse.cz/tests/4531 (`GRUB_PARAM=ima_policy=tcb;debug_pagealloc=on;foo=bar foo2=bar2`)
* http://quasar.suse.cz/tests/4528 (`GRUB_PARAM=ima_policy=tcb`)

* https://openqa.suse.de/tests/3731116 (sle-15-SP2-Online@**aarch64-virtio**, `GRUB_PARAM=ima_policy=tcb;debug_pagealloc=on;foo=bar`)
* https://openqa.suse.de/tests/3731098 (sle-15-SP2-Online@**aarch64-virtio**, `GRUB_PARAM=ima_policy=tcb`)
* https://openqa.suse.de/tests/3731099 (sle-12-SP3-Server-DVD-Incidents-Kernel@**ppc64le-virtio**, `GRUB_PARAM=ima_policy=tcb`)
* https://openqa.suse.de/tests/3731100 (sle-12-SP1-Server-DVD-Incidents-Kernel@64bit, `GRUB_PARAM=ima_policy=tcb`)

### CODE REVIEW

**Comment by pevik** [comment]
@mdoucha @cfconrad @richiejp FYI.

File: lib/bootloader_setup.pm [hunk 1]
```diff
@@ -99,48 +100,54 @@ And of course the new entries have C<ima_policy=tcb> added to kernel parameters.
...
-    my $section_new = "sed -e '1,/$script_new_esc/d' -e '/$script_new_esc/,\$d' " . GRUB_CFG_FILE;
     my $cnt_old     = script_output("$section_old | grep -c 'menuentry .$distro'");
-    my $cnt_new     = script_output("$section_new | grep -c 'menuentry .$distro'");
-    die("Unexpected number of grub entries: $cnt_new, expected: $cnt_old") if ($cnt_old != $cnt_new);
-    $cnt_new = script_output("grep -c 'menuentry .$distro.*($grub_param)' " . GRUB_CFG_FILE);
-    die("Unexpected number of new grub entries: $cnt_new, expected: " . ($cnt_old)) if ($cnt_old != $cnt_new);
-    $cnt_new = script_output("grep -c -E 'linux.*(/boot|/vmlinu[xz]-).* $grub_param ' " . GRUB_CFG_FILE);
-    die("Unexpected number of new grub entries with '$grub_param': $cnt_new, expected: " . ($cnt_old)) if ($cnt_old != $cnt_new);
+
+    my $i = 10;
```
Comment (cfconrad) [inline lib/bootloader_setup.pm]: Do we need `$i` anymore?

Comment (cfconrad) [reply lib/bootloader_setup.pm]: Ok, now I see it in line 130

File: lib/bootloader_setup.pm [hunk 2]
```diff
@@ -99,48 +100,54 @@ And of course the new entries have C<ima_policy=tcb> added to kernel parameters.
...
+    foreach my $grub_param (@grub_params) {
+        $i++;
+        my $script_new     = "/etc/grub.d/${i}_linux_openqa";
+        my $script_new_esc = $script_new =~ s~/~\\/~rg;
+        assert_script_run("cp -v $script_old $script_new");
```
Comment (cfconrad) [inline lib/bootloader_setup.pm]: Do we need to check for existing entries? If someone call this function on a hdd which have already extended boot entries?

File: lib/bootloader_setup.pm [hunk 3]
```diff
@@ -99,48 +100,54 @@ And of course the new entries have C<ima_policy=tcb> added to kernel parameters.
 
 =cut
 sub add_custom_grub_entries {
-    my $grub_param = get_var('GRUB_PARAM');
-    return unless defined($grub_param);
+    my @grub_params = split(/;/, get_var('GRUB_PARAM', ''));
```
Comment (cfconrad) [inline lib/bootloader_setup.pm]:
```suggestion
    my @grub_params = split(/\s*;\s*/, trim(get_var('GRUB_PARAM', '')));
```
using https://mojolicious.org/perldoc/Mojo/Util#trim

**Comment by cfconrad** [comment]
Nit comments above. 
LGTM

File: lib/bootloader_setup.pm (hunk 3 above)
Comment (pevik) [reply lib/bootloader_setup.pm]: @cfconrad thanks for a review. IMHO trim is not needed (everything works the same with extra spaces), but ok, I'll add it. 

I'd like to get this + https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9217 merged, I consider poo#61203 (and poo#51743 which is also partly affected by this) as a high priority.

**Comment by pevik** [comment]
@cfconrad trim added, can you please merge this one? BTW this and https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9217 has tiny conflict in `variables.md` (easy to rebase).

**Pull request merged**